### DISCUSSION TITLE
Suggestion: Boosters should work in both directions

### DISCUSSION THREAD

**Discussion by SilvariusCoyote** [OpenRCT2/OpenRCT2#15725] [Feature requests]
This is pretty self-explanatory. If you look at the behavior of brakes, they slow down trains going both backwards and forwards. Boosters do not currently exhibit this behavior; they speed up a train going forwards over the booster, but they do not affect the speed of a train going backwards over the booster. I beleive this behavior should be changed to allow boosters to affect the speed of trains going both backwards and forwards over them. This would allow us to create backwards launches without the use of shoestringing (provided the train enters the boosters going backwards) and make it easier to create more realistic forward-backward-forward launch coasters, such as Premier Skyloops and White Dragon in the Sky.

**Comment by Gymnasiast** [comment]
Is there _any_ chance of this alteration adversely affecting current designs?

**Comment by SilvariusCoyote** [comment]
I have seen a few examples of people making 'triple launch' coaster based on real-life designs.
Example:
![triplelaunchex](https://user-images.githubusercontent.com/19209095/31519374-de691214-af67-11e7-8831-b968a54b6f67.PNG)
This should hopefully show you a possible adversely affected design, based on White Dragon in the Sky. The coaster launches forward but doesn't clear the loop, travels backward over the launch without speeding up, then launches forward again, this time gaining enough speed to clear the loop and the rest of the layout. Adding backwards launch behavior would give the train too much speed when travelling backwards, possibly causing it to fly off the vertical spike. However, such designs could be fixed by merely adding to the height of the vertical spike.

**Comment by Gymnasiast** [comment]
Perhaps ask this on NE? I imagine most people building such a design are on NE anyway.

**Comment by X123M3-256** [comment]
This could well break designs that depend on this behaviour, but such designs are a relatively recent introduction since boosters didn't exist in vanilla, so there won't be too many around.

If you're building a triple launch then this is the behaviour you want - the train should accelerate on the reverse pass. On the flip side, if you're building a shuttle coaster then this may well not be what you want - in fact, you may wish for opposite - so I think changing this behaviour should be carefully considered.

**Comment by jensj12** [comment]
I have plenty of RCT1 designs that will break with this change, as well as a coaster in a group park.
Perhaps just wait until a seperate bidirectional booster piece can be added, or the existing piece gets more per-ride/per-piece configurable options.

**Comment by SilvariusCoyote** [comment]
What if we used the cable lift flag to set whether boosters work in multiple directions? Basically, booster track without the cable lift flag works the way it does currently, and boosters track with the cable lift flag works both forwards and backwards. When building booster track pieces there would be a checkbox on the GUI that decides whether booster pieces are built with or without this flag. It's a bit of a workaround but it's the best way I can think of with my rudimentary understanding of the .sv6 limitations.

**Comment by Gymnasiast** [comment]
How we (and RCTC) implemented boosters is already hacky to begin with, sharing its ID with the spinning control toggle track and having that guarded by lots of ifs. I'm not sure I want to pile another hack onto it by (ab)using the chain lift flag (not cable lift, that's something else), just to implement it within the SV6 constraints.

**Comment by SilvariusCoyote** [comment]
So is this something that should be saved for .opensv6?

**Comment by duncanspumpkin** [comment]
Yeah leave it for new file format.

**Comment by Ryder17z** [comment]
I think the best option is to have separate pieces, because am facing the same problem as @jensj12

**Comment by chaitanyathengdi** [comment]
How about adding a separate piece for forward and reverse boosters? I've provided an example of why I'd want a separate reverse booster(but NOT the same piece boosting in both directions) in the issue linked above.
I don't agree that boosters are similar to brakes and should work in both directions - brakes are based on friction which obviously works in both directions, but boosters are a different concept(think - maglev trains).
Of course, I'd imagine that if boosters were exactly the same as maglev trains then they'd always apply a force in a given direction and so act as a brake when the train passes in the opposite direction; but let's ignore that and treat it as not having an effect at all.

**Comment by duncanspumpkin** [comment]
Any new piece would require a new save format.

**Comment by chaitanyathengdi** [comment]
Okay - hopefully this'll be added when  the new save comes out :)

**Comment by guillaumecardon** [comment]
I'm also interested by this new piece (and the new save format is actually tested by the community).
There are more and more coaster with reverse section (Expedition Everest, Big Grizzly Mountain Runaway, Hagrid's motorbikes, Pegase Express etc.), i think it could be a nice feature to combine with the "shoestring" method

**Comment by spacek531** [comment]
NSF is getting closer - do we want to think about this again?

Here are some recent issues opened that discuss track features:

https://github.com/OpenRCT2/OpenRCT2/issues/15346
https://github.com/OpenRCT2/OpenRCT2/issues/15331
https://github.com/OpenRCT2/OpenRCT2/issues/15346
https://github.com/OpenRCT2/OpenRCT2/issues/15367

There's 6 different behaviors I can see being useful for boosters:

- forward boost, backward neutral
- forward boost, backward brake
- forward neutral, backward boost
- forward brake, backward boost
- forward boost, backward boost
- forward brake, backward brake

The brake feature would be similar to the reverse freefall coaster brake and mimic an eddy current brake.

**Comment by astraylife1** [comment]
With the new save format now implemented, can this become a reality?

Could we have two separate track pieces, one for forward boosters and one for backward?

In theory you could alternate each piece when building, forward, backward, forward, etc so that the launch will work both ways and it won't effect any existing rides that utilize boosters.

**Comment by karst** [comment]
@ZehMatt I feel this might be the correct place for discussion of different modes of the Booster (Speed Controller, LSM, LIM, Tyredrive)

**Reply by ZehMatt** [reply]
I've started this https://github.com/OpenRCT2/OpenRCT2/discussions/19413
